datamol/reactions/_reactions.py

```diff
@@ -0,0 +1,290 @@
+from typing import cast
+from typing import Union
+
+import os
+import io
+
+from loguru import logger
+import numpy as np
+
+import fsspec
+
+from rdkit.Chem import rdChemReactions
+
+import datamol as dm
+
+
+ATTACHING_RXN = rdChemReactions.ReactionFromSmarts("[*;h:1]>>[*:1][*]")
+
+
+def rxn_from_smarts(rxn_smarts: str) -> dm.ChemicalReaction:
+    """
+    Create a reaction from smarts
+
+    Args:
+        rxn_smarts:  Reaction SMARTS string
+
+    Returns:
+        Initilized reaction.
+    """
+    rxn = rdChemReactions.ReactionFromSmarts(SMARTS=rxn_smarts)
+    rxn.Initialize()
+    return rxn
+
+
+def rxn_to_smarts(rxn: dm.ChemicalReaction) -> str:
+    """Create a SMARTS from a reaction.
+
+    Args:
+        rxn: dm.ChemicalReaction object.
+
+    Returns:
+        SMARTS as string.
+    """
+    return rdChemReactions.ReactionToSmarts(reaction=rxn)
+
+
+def rxn_from_block(
```

Comment:
@zhu0619 it only accept a block string now. For using a filepath you have r`rxn_from_block_file`